### ISSUE TITLE
Memory Test

### ISSUE THREAD

**Issue by simoneberrino** [eduardocasino/k-1008-visable-memory-card-replica#2]
Hi Eduardo,

After powerup the screen already looks like this at startup, it seems it's not complete.

![power up screen](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/fa764630-02de-4e1d-a588-b457848b5508)

I tested the memory with the glitchwork memory test and in fact it doesn't work as it should

![memory test](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/62810306-0031-4826-bad7-ba6612c92d82)

The first time I ran the test it ended with this

![end memory test](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/8faa2add-ac18-44f4-8918-6f12b6d3af7d)

Here is the assembled board
The memories are K565PY1

![card assembled 2](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/63ac7f5c-ab8b-4412-840e-ecb52cbd7683)

**Comment by eduardocasino** [comment]
Hi, Simone. Is the test consistent? Does it always end at the same point (3601) and with the same expected (50) and read (51) values?

Those memories are 4x1 bit, and bit 0 of 0x3601 points directly to U42 (highlighted in green):

![k-1008-visable-memory-fail](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/37220276/09704824-18b6-471b-b461-b0c16cfc02a6)

Do you have any spares?

(Jeez, how I love those ceramics! :) )

**Comment by simoneberrino** [comment]
The test is not consistent.
I started it another 5 times and if by X you mean the end of the cycle, it completed it several times without giving me the 3601 error anymore.


https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/8fa773b8-3663-49f2-b4c2-546143c7c2c9

I have 2 other spare memories and I tried them on the U42 position and nothing changed.

But there's something new:

during one of these tests I noticed an increase in 5v amperes, slowly at first and then more and more quickly from 0.84 A up to 1.17 A and then I unplugged it after frying something

![Screen-2024-06-11_21-48-03](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/44d37f3d-ca58-4c71-b549-2d6f2848497b)

The 12V always remains stable.

I did another test by writing this program that brings the content to FF.
The k-1008 is mapped $2000-$3FFF

0002 A9 00
0004 85 00
0006 A9 20  <------ 0007 START ADDRESS OF K-1008
0008 85 01
000A A2 20
000C A0 00
000E A9 FF  <------- 000F content to be written to the memory locations
0010 91 00
0012 C8 D0
0014 FB E6
0016 01 CA
0018 D0 F6
001A 00

the program start from 0002

Then returning to $2000 i changed the content manually from FF to 00 up to $2015 and other locations changed as you can see

![IMG20240611222920](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/3d4c10df-6ac3-4c63-9043-c6a6d408084c)


locations changed above:

2020 7F
2021 FF
2022 7F
2023 FF
2024 7F
2025 FF
2026 7F
2027 FF
2028 7F
2029 FF
202A 7F
202B FF
202C 7F
202D FF
202E 7F
202F FF
2030 7F
2031 FF
2032 7F
2033 FF
2034 7F

locations changed further down:

3000 FF
3001 FB
3002 FF
3003 FB
3004 FF
3005 FB
3006 FF
3007 FB
3008 FF
3009 FB
300A FF
300B FB
300C FF
300D FB
300E FF
300F FB
3010 FF
3011 FB
3012 FF
3013 FB
3014 FF
3015 FB

**Comment by eduardocasino** [comment]
```
      AAAA AAAA AAAA AAAA    DDDD DDDD
      FEDC BA98 7654 3210    7654 3210		  
2000  0010 0000 0000 0000    0000 0000
2020  0010 0000 0010 0000    0111 1111 BAD
2001  0010 0000 0000 0001    0000 0000
2021  0010 0000 0010 0001    1111 1111 OK

2000  0010 0000 0000 0000    0000 0000
3000  0011 0000 0000 0000    1111 1111 OK
2001  0010 0000 0000 0001    0000 0000
3001  0011 0000 0000 0001    1111 1011 BAD
```

I see two patterns here:
* In even addresses, no matter if A5 is high or low, bit 7 is selected. This points to a possible bad U21 chip.
* In odd addresses, no matter if A12 is high or low, bit 2 is selected. In this case, U43 may be bad.

To verify if I'm correct, please swap chips U21 and U43 and repeat the last test. In this case, you should see that the addresses that change to 7F are 2100, 2102, 2104, 2106...  and the ones that change to FB are 2021, 2023, 2025, 2027...

**Comment by simoneberrino** [comment]
It's correct!
I swapped the chips and the contents changed as you expected

![IMG20240612221810](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/bdeab21c-6bf9-4087-924b-2593ed333f9b)


here is the content from the serial monitor:

201F FF
2020 FF
2021 FB
2022 FF
2023 FB
2024 FF
2025 FB
2026 FF
2027 FB
2028 FF
2029 FB
202A FF
202B FB
202C FF
202D FB
202E FF
202F FB
2030 FF
2031 FB
2032 FF
2033 FB
2034 FF
2035 FB
2036 FF
2037 FF
2038 FF 2100
2100 7F
2101 FF
2102 7F
2103 FF
2104 7F
2105 FF
2106 7F
2107 FF
2108 7F
2109 FF
210A 7F
210B FF
210C 7F
210D FF
210E 7F
210F FF
2110 7F
2111 FF
2112 7F
2113 FF
2114 7F
2115 FF
2116 FF

**Comment by eduardocasino** [comment]
Then, at least those two chips are bad.

**Comment by simoneberrino** [comment]
I can confirm that it was U21 and U43 that were causing the problem.
I replaced them with the other 2 memories I had in addition and now there is another working K-1008 card!
I tested the memories again with the Glitch's memory test and they are ok.
As you can see here the image loaded with bad U21 and U43 and then the same image with the good chips.

![IMG20240612005308](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/e21c71de-8a11-46e0-9e39-c917b758feff)

![IMG20240613004338](https://github.com/eduardocasino/k-1008-visable-memory-card-replica/assets/172144720/8d97a2bc-84f9-493f-8f6d-661d154c1e68)

The thing I can't explain is that strange increase in ampere absorption at 5v when I did the test, it only happened once, however very strange

**Comment by eduardocasino** [comment]
Glad you got it working!

That increase could be caused by the faulty chips.

**Comment by simoneberrino** [comment]
Well, I'll keep an eye on it but I'd say the problem has been resolved at this point.

One last thing I would like to ask you:
for the tests I used this small monitor that can capture the NTSC signal, but I found a Sony LMD-1420 which is definitely better and that's what you used too if I'm not mistaken.
What type of connection did you use?
Just a video cable that goes to the BNC of the Sony or other special connections or settings?
